WcaOnRails/app/webpacker/components/RolesTab/DelegateForm.jsx

```diff
@@ -7,10 +7,24 @@ export default function DelegateForm({
   formValues,
   updateFormProperty,
   regions,
+  subRegions,
   delegateStatusOptions,
 }) {
   const handleFormChange = (_, { name, value }) => updateFormProperty({ [name]: value });
 
+  const subRegionsOfSelectedRegion = React.useMemo(() => {
+    let subregionsList = [];
+    if (!formValues.regionId) return [];
+    subregionsList = subRegions[formValues.regionId] || [];
+    if (subregionsList.length > 0) {
+      subregionsList.push({
+        name: 'None',
+        value: null,
+      });
+    }
+    return subregionsList;
+  }, [formValues.regionId, subRegions]);
+
```

Comment:
Not a big fan of `let` in JS. How about the following flow?

```suggestion
  const subRegionsOfSelectedRegion = React.useMemo(() => {
    if (!formValues.regionId) return [];

    const subregionsList = subRegions[formValues.regionId] || [];

    if (subregionsList.length > 0) {
      return [...subregionsList, {
        name: 'None',
        value: null,
      }];
    }

    return subregionsList;
  }, [formValues.regionId, subRegions]);
```